src/FrontDoor/FrontDoor/Cmdlets/SetAzureRmFrontDoor.cs

```diff
@@ -104,23 +108,36 @@ public class SetAzureRmFrontDoor : AzureFrontDoorCmdletBase
         /// Whether to enable use of this rule.
         /// </summary>
         [Parameter(Mandatory = false, HelpMessage = "Operational status of the Front Door load balancer. Default value is Enabled")]
+        [PSArgumentCompleter("Enabled", "Disabled")]
         public PSEnabledState EnabledState { get; set; }
 
         /// <summary>
         /// Whether to enforce certificate name check on HTTPS requests to all backend pools. No effect on non-HTTPS requests.
         /// </summary>
-        [Parameter(Mandatory = false, HelpMessage = "Whether to disable certificate name check on HTTPS requests to all backend pools. No effect on non-HTTPS requests.")]
+        [Parameter(Mandatory = true, ParameterSetName = FieldsWithCertificateNameCheckParameterSet, HelpMessage = "Whether to disable certificate name check on HTTPS requests to all backend pools. No effect on non-HTTPS requests.")]
+        [Parameter(Mandatory = true, ParameterSetName = ObjectWithCertificateNameCheckParameterSet, HelpMessage = "Whether to disable certificate name check on HTTPS requests to all backend pools. No effect on non-HTTPS requests.")]
+        [Parameter(Mandatory = true, ParameterSetName = ResourceIdWithCertificateNameCheckParameterSet, HelpMessage = "Whether to disable certificate name check on HTTPS requests to all backend pools. No effect on non-HTTPS requests.")]
         public SwitchParameter DisableCertificateNameCheck { get; set; }
```

Comment:
OK after some testing I've found the problem.
`DisableCertificateNameCheck` *was* in every parameter sets, but now it only exists in 3 of them -- this is a breaking change. That's why the static analysis was complaining.
Besides, a switch parameter shouldn't be mandatory. It doesn't make sense.
My suggestion is to change the `[Parameter]` back. Tell the user when does it take effect in the help.

Reply:
Thank you very much for helping pin down the issue! 

Is there a way to add it as an exception to the static analysis? The two parameters DisableCertificateNameCheck and BackendPoolsSettings really should be mutually exclusive because CertificateNameCheck is a field of BackendPoolsSettings. If the user passes in one value for DisableCertificateNameCheck, and a different value for BackendPoolsSettings, I'm not sure the best way to resolve that. Maybe I should add the parameter back, and then have a check that the two values are the same, and fail the operation if it isn't? It seems like it may result in a poorer user experience.

Reply:
We only add exceptions when it's a false positive, but this was a real breaking change, so we can't do it that way.

On a second thought, you should have 6 parameter sets instead of 9

|old parameter set name|new|
|-|-|
|ByFieldsParameterSet (Default)|ByFieldsParameterSet (Default)|
|ByObjectParameterSet|ByObjectParameterSet|
|ByResourceIdParameterSet|ByResourceIdParameterSet|
||ByFieldsWithBackendPoolsSettingParameterSet|
||ByObjectWithBackendPoolsSettingParameterSet|
||ByResourceIdWithBackendPoolSettingsParameterSet|

Make `-DisableCertificateNameCheck` in the first 3 and `-BackendPoolsSetting` in the last three. Then you are not making a breaking change and the two parameters are exclusive

Reply:
@isra-fel Having the 6 parameter sets as described above doesn't seem to work for all cases. Specifically, in the case where neither BackendPoolsSetting parameter nor DisableCertificateSource parameter are present, PowerShell throws the following error: "Parameter set cannot be resolved using the specified named parameters. One or more parameters issued cannot be used together or an insufficient number of parameters were provided." This is visible in the failure of TestFrontDoorCrudWithPiping test. 

When I first opened the PR, I had 9 parameter sets composed of (ByFields, ByObject, ByResourceId) x (BackendPoolsSetting present, DisableCertificateNameCheck present, neither present). It also looks like all checks and tests pass if I keep the original parameter sets and just add another parameter BackendPoolsSetting, and then maybe in the code check if DisableCertificateCheck is also present, and manually fail or override one value with the other.

How do you suggest I proceed with this issue? Many thanks!

Reply:
Followed up with @isra-fel offline to explain how 9 parameter sets were designed to be so.